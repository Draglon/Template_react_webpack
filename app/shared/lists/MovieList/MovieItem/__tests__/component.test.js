import React from 'react';

import MovieItemComponent from '../component';

describe('<MovieItemComponent /> container', () => {
  let component;
  let props = {
    item: {
      id: 'some id',
      title: 'some title',
      overview: 'some overview',
      poster_path: 'path to image',
    },
  };

  it('Render property with card actions', () => {
    props = {
      ...props,
      actions: [1, 2, 3],
    };
    component = shallow(<MovieItemComponent {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Render snapshot - without card actions', () => {
    props = {
      ...props,
      actions: null,
    };
    component = shallow(<MovieItemComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
