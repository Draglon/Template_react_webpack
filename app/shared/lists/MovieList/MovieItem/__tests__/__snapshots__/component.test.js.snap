// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItemComponent /> container Render property with card actions 1`] = `
<Card
  actions={
    Array [
      1,
      2,
      3,
    ]
  }
  className="top-margin"
  cover={
    <ImageComponent
      alt="some title"
      className="movie__img"
      path="path to image"
    />
  }
  hoverable={true}
>
  <Link
    replace={false}
    to="/movie/some id"
  >
    <Meta
      className="text-size"
      description="some overview"
      title="some title"
    />
  </Link>
</Card>
`;

exports[`<MovieItemComponent /> container Render snapshot - without card actions 1`] = `
<Card
  actions={null}
  className="top-margin"
  cover={
    <ImageComponent
      alt="some title"
      className="movie__img"
      path="path to image"
    />
  }
  hoverable={true}
>
  <Link
    replace={false}
    to="/movie/some id"
  >
    <Meta
      className="text-size"
      description="some overview"
      title="some title"
    />
  </Link>
</Card>
`;
